### PR TITLE
Minor docs updates for the week

### PR DESCRIPTION
**Pull request by sydney-runkle**
Fix https://github.com/pydantic/pydantic/issues/8771
Fix https://github.com/pydantic/pydantic/issues/8736

Selected Reviewer: @lig

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Please review

File: docs/api/standard_library_types.md [hunk 1]
```diff
@@ -59,7 +59,9 @@ types:
       (if < `-2e10`or > `2e10`) since 1 January 1970
     * `str`; the following formats are accepted:
         * `YYYY-MM-DD[T]HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM]`
+        * `YYYY-MM-DD` is accepted in lax mode, but not in strict mode
         * `int` or `float` as a string (assumed as Unix time)
+    * [`datetime.date`][] instances are accepted in lax mode, but not in strict mode
```
Comment (davidhewitt) [inline docs/api/standard_library_types.md:64]: Is this `[]` missing a link?

**Review by davidhewitt — APPROVED** [review]
Lgtm with just one question!

**Review by hramezani — APPROVED** [review]

File: docs/api/standard_library_types.md (hunk 1 above)
Comment (sydney-runkle) [reply docs/api/standard_library_types.md:64]: Good question! We're using new sphinx links that work with this syntax to the stdlib docs :). So in short, no. You can see this active link here :) https://74596f4b.pydantic-docs2.pages.dev/api/standard_library_types/#datetimedatetime

Comment (sydney-runkle) [reply docs/api/standard_library_types.md:64]: @Viicos,

Your docs work is coming in handy :).

Comment (Viicos) [reply docs/api/standard_library_types.md:64]: Yes actually ``[`datetime.date`][]`` is a shortcut for ``[`datetime.date`][datetime.date]``

**Pull request merged**